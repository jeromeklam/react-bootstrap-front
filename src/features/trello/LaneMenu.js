import React from 'react';
import PropTypes from 'prop-types';
import { DropdownWrapper } from '../advanced';
import { DropdownMenu, DropdownMenuOption, DropdownMenuDivider } from '../basic';

const LaneMenu = props => {
  const refOptions = React.createRef();
  return (
    <>
      <DropdownWrapper
        position="bottom"
        trigger={<button className="btn btn-sm btn-light trello-lane-menu-menu">&hellip;</button>}
        tooltip="Actions Ã  effectuer"
      >
        <DropdownMenu className="trello-lane-menu">
          {props.options &&
            props.options.length > 0 &&
            props.options.map(option => (
              <DropdownMenuOption
                key={option.label}
                onClick={() => option.onClick(props.id, refOptions)}
                label={option.label}
                theme={option.theme || null}
              />
            ))}
          {props.delete && <DropdownMenuDivider />}
          {props.delete && (
            <DropdownMenuOption
              onClick={props.onDelete}
              label={props.t({ id: 'rbf.trello.lane.delete', defaultMessage: 'Delete lane' })}
              theme="warning"
            />
          )}
        </DropdownMenu>
      </DropdownWrapper>
      <div ref={refOptions} />
    </>
  );
};

LaneMenu.propTypes = {
  id: PropTypes.string.isRequired,
  delete: PropTypes.bool,
  onDelete: PropTypes.func,
  options: PropTypes.array,
  t: PropTypes.func.isRequired,
};

LaneMenu.defaultProps = {
  delete: false,
  onDelete: () => {},
  options: [],
};

export default LaneMenu;
