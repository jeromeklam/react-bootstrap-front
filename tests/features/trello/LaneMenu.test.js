import React from 'react';
import renderer from 'react-test-renderer';
import LaneMenu from '../../../src/features/trello/CardMenu';

const callback = jest.fn();

it('renders standard', () => {
  // First render
  const component = renderer.create(
    <LaneMenu t={callback} />
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});
