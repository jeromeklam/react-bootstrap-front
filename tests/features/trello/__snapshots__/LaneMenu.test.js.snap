// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders standard 1`] = `
<div
  className="advanced-dropdown-wrapper"
  onClick={[Function]}
  title="Actions à effectuer sur la carte"
>
  <button
    className="btn btn-sm btn-light trello-card-menu-menu"
  >
    …
  </button>
</div>
`;
